File: Frontend/lib/models/vet.dart
import 'package:pet_connect_app/models/user.dart';

class Vet {
  final String id;
  final String phone;
  final String address;
  final String specialization;
  final int yearsOfExperience;
  final User user;

  Vet({
    required this.id,
    required this.phone,
    required this.address,
    required this.specialization,
    required this.yearsOfExperience,
    required this.user,
  });

  factory Vet.fromJson(Map<String, dynamic> json) {
    return Vet(
      id: json['id'],
      phone: json['phone'],
      address: json['address'],
      specialization: json['specialization'],
      yearsOfExperience: json['yearsOfExperience'],
      user: User.fromJson(json['user']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'phone': phone,
      'address': address,
      'specialization': specialization,
      'yearsOfExperience': yearsOfExperience,
      'user': user.toJson(),
    };
  }
}

File: backend_java/src/main/java/com/petconnect/model/Vet.java
package com.petconnect.model;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.MapsId;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@Entity
@Table(name = "vets")
public class Vet {

    @Id
    private String id;

    private String phone;
    private String address;
    private String specialization;
    private int yearsOfExperience;

    @OneToOne
    @MapsId
    @JoinColumn(name = "id")
    private User user;

    public Vet() {
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getSpecialization() {
        return specialization;
    }

    public void setSpecialization(String specialization) {
        this.specialization = specialization;
    }

    public int getYearsOfExperience() {
        return yearsOfExperience;
    }

    public void setYearsOfExperience(int yearsOfExperience) {
        this.yearsOfExperience = yearsOfExperience;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}
